### PR TITLE
Package libqfieldsync as wheel

### PR DESCRIPTION
**Pull request by m-kuhn**
Depends on
- https://github.com/opengisch/qgis-plugin-ci/pull/230
- https://github.com/opengisch/libqfieldsync/pull/49

Developer information, to work locally, either install the `requirements.txt` (if you don't need to edit libqfieldsync) or pip install libqfieldsync as editable if you need to work on libqfieldsync.

```sh
git clone git@github.com:opengisch/libqfieldsync.git
pip install -e libqfieldsync
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,3 +4,6 @@ flake8-respect-noqa
 pytest
 future
 transifex-client
+
+# TODO: point to a commit on master after merging https://github.com/opengisch/libqfieldsync/pull/49
```
Comment (m-kuhn) [inline requirements.txt:8]: Open todo ...

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -89,6 +90,10 @@ jobs:
         run: |
           sudo apt update && sudo apt install qtbase5-dev qttools5-dev-tools
           sudo pip install qgis-plugin-ci
+      - name: Package libqfieldsync
+        run: |
+          pip wheel $(grep -o -P '(https://.*.tar.gz)' requirements.txt)
```
Comment (suricactus) [inline .github/workflows/continuous_integration.yml:95]:
```suggestion
          pip wheel $(grep -o -P '(https://.*\.tar\.gz)' requirements.txt)
```

File: qfieldsync/__init__.py [hunk 3]
```diff
@@ -25,9 +25,17 @@
...
 
 import os
+import pathlib
+import sys
 
 import qgis.utils
 
+src_dir = pathlib.Path(__file__).parent.resolve()
+
+libqfieldsync_whl = src_dir / "libqfieldsync.whl"
```
Comment (suricactus) [inline qfieldsync/__init__.py:35]:
```suggestion
libqfieldsync_whl = src_dir.joinpath("libqfieldsync.whl")
```
I always feel afraid of too much magic with operator overload.

**Comment by m-kuhn** [comment]
@suricactus this should be ready.
One TODO: point to a commit on master after merging the PR on https://github.com/opengisch/libqfieldsync/pull/49

**Comment by m-kuhn** [comment]
I think we will want to points to a sha on master and update explicitly

**Comment by suricactus** [comment]
Yes, we want that for releases, I do have it mind. Will pinpoint when I am about to release a new version, there is a bit of other development to be made for now.

Also thinking whether to create a `release` branch as I did on QFieldCloud and pinpoint only there, so we are forced to keep libqfieldsync and qfieldsync master branches compatible with each other, until we think it's not working.

**Pull request merged**